### ISSUE TITLE
Run `R CMD check` on Windows and macOS as well as Ubuntu in CI

### ISSUE THREAD

**Issue by dylanhmorris** [CDCgov/ww-inference-model#189]
> This suggests it would be beneficial to run `R CMD check` in CI on macOS and `Windows as well as Ubuntu. I will create an issue.

_Originally posted by @dylanhmorris in https://github.com/CDCgov/ww-inference-model/issues/187#issuecomment-2379798202_

We will need to address #145, since the current snapshot test of MCMC output is architecture-specific.

**Comment by gvegayon** [comment]
@kaitejohnson, I think you are already addressing this in https://github.com/CDCgov/ww-inference-model/tree/189-add-windows-check?

**Comment by kaitejohnson** [comment]
Yes except it doesn't work! I think that install of cmdstanr on windows won't work. Perhaps there is a workaround, or perhaps we close this... https://www.maxmantei.com/2020/05/16/cmdstanr-windows/

In EpiNow2, they don't install cmdstanr on windows in R CMD check bc I think it doesn't work https://github.com/epiforecasts/EpiNow2/blob/ba7b48d10a3989580642e73f72087d7d76f2f4ea/.github/workflows/R-CMD-check.yaml#L69

**Comment by gvegayon** [comment]
It seems their script is failing to fetch the latest version of the package [here](https://github.com/epinowcast/actions/blob/aa353d813acf9ac54e4fe23494d019dba01e917b/install-cmdstan/scripts/get-latest-release.ps1#L19-L28). @seabbs, what about using `gh cli` directly? This would do the same thing:

```bash
gh api \
    -H "Accept: application/vnd.github+json" \
    -H "X-GitHub-Api-Version: 2022-11-28" \
    repos/stan-dev/cmdstan/releases/latest
``` 

And it would work across OSs.

**Comment by seabbs** [comment]
I think it could be good to use `gh` (though it adds more weight to action) but I am not sure that will solve the issue resolving the latest version. Worth a try though.

**Comment by gvegayon** [comment]
`gh cli` comes shipped with the runners. The only issue would be if you run it within a container.

**Comment by seabbs** [comment]
your right! I implemented this today and it looks good to me

**Comment by gvegayon** [comment]
Closing this one as @kaitejohnson and @seabbs addressed Windows in #190 and OSX is now a separate issue in #204.
